archive/frames/utils.py

```diff
@@ -156,3 +157,61 @@ def build_nginx_zip_text(frames, directory, uncompress=False):
 
     # Concatenate all lines together into a single string
     return ''.join(ret)
+
+
+def aggregate_raw_sql(frames, timeout=0):
+    frames = frames.values_list(
+        "proposal_id",
+        "site_id",
+        "telescope_id",
+        "instrument_id",
+        "configuration_type",
+        "primary_optical_element",
+    ).order_by()
+
+    with connections["replica"].cursor() as cursor:
+        filtered_sql, params = frames.query.sql_with_params()
+        params = (timeout,) + params
+        query_sql = f"""
+          SET LOCAL statement_timeout TO %s;
```

Comment:
If you could add a comment in this method for future me to have some greater understanding of how you came up with this sql that would be awesome. Something about why you picked the group index you did, and what the important things you learned in building this query so we don't forget it...